src/main/webapp/calendar.html

```diff
@@ -38,5 +40,24 @@
   </div>
 
 </body>
+<script>
+  var lastUsedId = "";
+  $('body').on('click', function (e) {
+    if (!isIdValid(lastUsedId)) {
+      if (document.getElementsByClassName("popover").length)
+        lastUsedId = document.getElementsByClassName("popover")[0].id;
+    }
+    else {
+      if (e.target.id === lastUsedId || e.target.parentElement.id === lastUsedId || hasAnyParentWithGivenId(e.target.parentElement, lastUsedId))
+        return;
+      $('#' + lastUsedId).popover('hide');
+      if (document.getElementsByClassName("popover").length == 1)
+        lastUsedId = document.getElementsByClassName("popover")[0].id;
+      else if (document.getElementsByClassName("popover").length == 2)
+        lastUsedId = document.getElementsByClassName("popover")[1].id;
+      else lastUsedId = "";
```

Comment:
What is the purpose of this section? This might be more readable with a high-level comment.

Reply:
Done